### PR TITLE
Option to set ServerSideVerificationOptions for new RewardedAd Api

### PR DESCRIPTION
**Pull request by halfdevil**
Option to set user id and custom reward string for new rewarded ad api. This is required for passing user id to google for server side verification.

### CODE REVIEW

File: source/plugin/Assets/Plugins/iOS/GADUServerSideVerificationOptions.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+//  Google Mobile Ads SDK
+//
+//  Copyright 2018 Google Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <GoogleMobileAds/GoogleMobileAds.h>
+
+/// Options for server-to-server verification callbacks for a rewarded ad.
+@interface GADUServerSideVerificationOptions : NSObject
```
Comment (stowy) [inline source/plugin/Assets/Plugins/iOS/GADUServerSideVerificationOptions.h]: is this class adding anything that isn't already available from the `GADServerSideVerificationOptions` class?  It looks a bit redundant to me. Why not remove it?

Comment (halfdevil) [reply source/plugin/Assets/Plugins/iOS/GADUServerSideVerificationOptions.h]: Done. I have added that as a wrapper class, it is not required.

**Comment by stowy** [comment]
Internal checks will always fail due to the meta file.

File: source/plugin/Assets/GoogleMobileAds/Api/ServerSideVerificationOptions.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018 Google, Inc.
```
Comment (stowy) [inline source/plugin/Assets/GoogleMobileAds/Api/ServerSideVerificationOptions.cs]: 2019 please.

File: source/plugin/Assets/Plugins/iOS/GADUInterface.m [hunk 3]
```diff
@@ -391,6 +390,27 @@ void GADUTagForChildDirectedTreatment(GADUTypeRequestRef request, BOOL childDire
   internalRequest.tagForChildDirectedTreatment = childDirectedTreatment;
 }
 
+/// Creates an empty GADUServerSideVerificationOptions and returns its reference.
```
Comment (stowy) [inline source/plugin/Assets/Plugins/iOS/GADUInterface.m]: GADServerSideVerificationOptions

File: source/plugin/Assets/Plugins/iOS/GADUInterface.m [hunk 4]
```diff
@@ -391,6 +390,27 @@ void GADUTagForChildDirectedTreatment(GADUTypeRequestRef request, BOOL childDire
...
+GADUTypeServerSideVerificationOptionsRef GADUCreateServerSideVerificationOptions() {
+  GADServerSideVerificationOptions *options = [[GADServerSideVerificationOptions alloc] init];
+  GADUObjectCache *cache = [GADUObjectCache sharedInstance];
+  [cache.references setObject:options forKey:[options gadu_referenceKey]];
+  return (__bridge GADUTypeServerSideVerificationOptionsRef)(options);
+}
+
+/// Sets the user id on the GADUServerSideVerificationOptions
```
Comment (stowy) [inline source/plugin/Assets/Plugins/iOS/GADUInterface.m]: GADServerSideVerificationOptions

File: source/plugin/Assets/Plugins/iOS/GADUInterface.m [hunk 5]
```diff
@@ -391,6 +390,27 @@ void GADUTagForChildDirectedTreatment(GADUTypeRequestRef request, BOOL childDire
...
+void GADUServerSideVerificationOptionsSetUserId(GADUTypeServerSideVerificationOptionsRef options, const char *userId) {
+  GADServerSideVerificationOptions *internalOptions = (__bridge GADServerSideVerificationOptions*)options;
+  internalOptions.userIdentifier = GADUStringFromUTF8String(userId);
+}
+
+/// Sets the custom reward string on the GADUServerSideVerificationOptions
```
Comment (stowy) [inline source/plugin/Assets/Plugins/iOS/GADUInterface.m]: GADServerSideVerificationOptions

File: source/plugin/Assets/Plugins/iOS/GADUInterface.m [hunk 6]
```diff
@@ -484,6 +504,13 @@ void GADURequestNativeAd(GADUTypeAdLoaderRef adLoader, GADUTypeRequestRef reques
   [internalAdLoader loadRequest:[internalRequest request]];
 }
 
+/// Sets the GADUServerSideVerificationOptions on GADURewardedAd
```
Comment (stowy) [inline source/plugin/Assets/Plugins/iOS/GADUInterface.m]: GADServerSideVerificationOptions

**Comment by stowy** [comment]
Thanks @halfdevil 👍

**Pull request merged**